packages/server/test/with-blitz.test.ts

```diff
@@ -2,11 +2,29 @@ import {withBlitz} from '../src'
 
 describe('withBlitz', () => {
   it('alters the webpack config as expected', () => {
-    const nextConfigFn = withBlitz({})
+    const nextConfigFn = withBlitz({
+      foo: 'bar',
+    })
     const newNext = nextConfigFn('', {defaultConfig: {}})
     const newNextWithoutWebpack = Object.assign({}, newNext, {webpack: null})
 
     expect(newNextWithoutWebpack).toStrictEqual({
+      foo: 'bar',
+      experimental: {
+        reactMode: 'concurrent',
+      },
+      webpack: null,
+    })
+  })
+
+  it('can handle functional next configs', () => {
+    const nextConfigFn = withBlitz((phase: any) => ({phase, foo: 'bar'}))
+    const newNext = nextConfigFn('foo', {defaultConfig: {}})
+    const newNextWithoutWebpack = Object.assign({}, newNext, {webpack: null})
+
+    expect(newNextWithoutWebpack).toStrictEqual({
+      phase: 'foo',
+      foo: 'bar',
       experimental: {
         reactMode: 'concurrent',
       },
```

Comment:
Should `withBlitz` return a function here since the input is a function?

Reply:
`withBlitz` always returns a function because it uses `withPlugins` when you call the function you flatten the config to an object. So from `newNext` onwards we have an object.